### ISSUE TITLE
[DocDB] Adjust GC of schema packings to account for XCluster 

### ISSUE THREAD

**Issue by lingamsandeep** [yugabyte/yugabyte-db#17229]
Jira Link: [DB-6479](https://yugabyte.atlassian.net/browse/DB-6479)
### Description

For XCluster integration with packed rows, the producer schema version is overwritten with a compatible consumer schema version upon receiving rows via replication.

However, there is one issue that exists with the GC of older schema packings.

Consider the example where replication is setup from A (Schema version 1) ->B (Schema version 1) and replication is in a steady state, but no data has been written yet.

- Replication is unhealthy
- Customer writes several rows to A with schema version of 1.
- Customer performs schema change on B. Schema version on B: 2
- Compaction happens on B, so schema packings for schema_version 1 are GC’ed
- Replication now becomes healthy and we have rows flowing in with schema version of 1, but we don’t have the corresponding schema_packing 1 because it has been GC’ed.

Basically, there is an edge case if the replication is unhealthy and consumer performs GC of the schema packings because it has not even received any rows yet, then we could have a problem. 

Potential solutions:

- When replication is healthy and fully caught up, it should be safe to GC all older schema packings.
- When replication is unhealthy, we need to retain all older schema packings until the XCluster retention period of 24hr.

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-6479]: https://yugabyte.atlassian.net/browse/DB-6479?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by rthallamko3** [comment]
Per Sandeep, xCluster metadata has the schema version dependencies between the source and target, so if the GC can be restricted to the ones not in the purview of xCluster, that can help with this issue. cc @Huqicheng

**Comment by rthallamko3** [comment]
@lingamsandeep , Do we want to backport to 2.18 in case any xCluster customers want it?

**Comment by lingamsandeep** [comment]
Backported to 2.18.
